src/hotspot/cpu/aarch64/stubGenerator_aarch64.cpp

```diff
@@ -3417,15 +3417,15 @@ class StubGenerator: public StubCodeGenerator {
     Register rscratch3 = r10;
     Register rscratch4 = r11;
 
-    __ andw(rscratch3, r2, r4);
-    __ bicw(rscratch4, r3, r4);
     reg_cache.extract_u32(rscratch1, k);
     __ movw(rscratch2, t);
-    __ orrw(rscratch3, rscratch3, rscratch4);
     __ addw(rscratch4, r1, rscratch2);
```

Comment:
Can you try to replace these 2 lines with following?
```
    __ movw(rscratch4, t);
    __ addw(rscratch4, r1, rscratch4);
```

I expect it could bring more performance gain, but not sure.

Reply:
Thanks! I can't measure any difference at all with that change, seems to perform identically.

Reply:
I see, Thanks!

Reply:
Unless you really want zero extension it's better to use `mov` than `movw` (or `orrw`). 
On many AArch64 implementations `mov` doesn't even issue. Instead, it is handled by the renamer during the decode stage. However, because it has to clear the upper 32 bits, `movw` does issue.

Reply:
Thanks! I don't measure a throughput improvement (didn't check any perf counters like instructions retired) when changing the `movw` to `mov` for `t` across F, G, H and I on Neoverse N1 or Mac M1. I'm also not sure how to tell if it's safe, my knowledge is shallow here.